agent/src/main/java/com/teamscale/jacoco/agent/options/sapnwdi/DelayedSapNwdiMultiUploader.java

```diff
@@ -0,0 +1,71 @@
+package com.teamscale.jacoco.agent.options.sapnwdi;
+
+import com.teamscale.client.CommitDescriptor;
+import com.teamscale.jacoco.agent.upload.IUploader;
+import com.teamscale.jacoco.agent.util.LoggingUtils;
+import com.teamscale.report.jacoco.CoverageFile;
+import org.slf4j.Logger;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.Executor;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+/**
+ * Wraps an {@link IUploader} and in order to delay upload until a {@link CommitDescriptor} is asynchronously made
```

Comment:
Actually this seems to be not "an" but multiple uploaders. Could you also try to explain what happens if some commits are already available, but others are not? I guess in this case, the respective applications are just not deployed and therefore cannot have any coverage anyway. Correct?

Reply:
Sure